Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -1,129 +1,153 @@
-import { projectContainer } from '../design/styles';
-import {
-  dispatcherState,
-  settingsState,
-  botDisplayNameState,
-  publishTypesState,
-  publishHistoryState,
-} from '../../recoilModel';
-import { navigateTo } from '../../utils/navigation';
+import { dispatcherState, botProjectSpaceSelector } from '../../recoilModel';
 import { Toolbar, IToolbarItem } from '../../components/Toolbar';
-import { OpenConfirmModal } from '../../components/Modal/ConfirmDialog';
+import { createNotification } from '../../recoilModel/dispatchers/notification';
+import { Notification } from '../../recoilModel/types';
+import { getSensitiveProperties } from '../../recoilModel/dispatchers/utils/project';
 
-import { TargetList } from './targetList';
 import { PublishDialog } from './publishDialog';
-import { ContentHeaderStyle, HeaderText, ContentStyle, contentEditor, overflowSet, targetSelected } from './styles';
-import { CreatePublishTarget } from './createPublishTarget';
-import { PublishStatusList, IStatus } from './publishStatusList';
+import { ContentHeaderStyle, HeaderText, ContentStyle, contentEditor } from './styles';
+import { IStatus } from './publishStatusList';
+import { BotStatusList, IBotStatus } from './botStatusList';
+import { pendingNotificationCard, publishedNotificationCard } from './notifications';
 import { PullDialog } from './pullDialog';
 
 const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: string }>> = (props) => {
-  const selectedTargetName = props.targetName;
   const { projectId = '' } = props;
-  const [selectedTarget, setSelectedTarget] = useState<PublishTarget | undefined>();
-  const settings = useRecoilValue(settingsState(projectId));
-  const botName = useRecoilValue(botDisplayNameState(projectId));
-  const publishTypes = useRecoilValue(publishTypesState(projectId));
-  const publishHistory = useRecoilValue(publishHistoryState(projectId));
-
+  const botProjectsMeta = useRecoilValue(botProjectSpaceSelector);
   const {
+    getPublishHistory,
     getPublishStatus,
     getPublishTargetTypes,
-    getPublishHistory,
     setPublishTargets,
     publishToTarget,
     setQnASettings,
     rollbackToVersion: rollbackToVersionDispatcher,
+    addNotification,
+    deleteNotification,
     setCurrentPageMode,
   } = useRecoilValue(dispatcherState);
 
-  const [addDialogHidden, setAddDialogHidden] = useState(true);
-  const [editDialogHidden, setEditDialogHidden] = useState(true);
+  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
+
+  const [showNotifications, setShowNotifications] = useState<{ [key: string]: boolean }>({});
+  // fill Settings, status, publishType, publish target for bot from botProjectMeta
+  const botsettingList: { [key: string]: any }[] = [];
+  const statusList: IBotStatus[] = [];
+  const botPublishTypesList: { [key: string]: any }[] = [];
+  const publishHistoyList: { [key: string]: any }[] = [];
+  const publishTargetsList: { [key: string]: any }[] = [];
```

Comment:
Instead of any we should use {projectId: string; publishTargets: PublishTarget\[\]}?